lib/internal/streams/pipeline.js

```diff
@@ -81,9 +87,15 @@ function pipeline(...streams) {
     const writing = i > 0;
     return destroyer(stream, reading, writing, function(err) {
       if (!error) error = err;
-      if (err) destroys.forEach(call);
+      if (err) {
+        for (const destroy of destroys) {
+          destroy(err);
+        }
+      }
       if (reading) return;
-      destroys.forEach(call);
+      for (const destroy of destroys) {
+        destroy();
+      }
```

Comment:
Thinking out loud, are these destroy loops `O(n^2)`? i.e. each `destroy` will call each `destroyer` which will call each `destroy` until every `once` inside of `eos` have been hit?